geowebcache/azureblob/src/main/java/org/geowebcache/azure/DeleteManager.java

```diff
@@ -169,14 +172,21 @@ public void issuePendingBulkDeletes() throws StorageException {
 
         try {
             Properties deletes = client.getProperties(pendingDeletesKey);
+            Set<String> deletesToClear = new HashSet<>();
             for (Map.Entry<Object, Object> e : deletes.entrySet()) {
                 final String prefix = e.getKey().toString();
                 final long timestamp = Long.parseLong(e.getValue().toString());
                 log.info(
                         String.format(
                                 "Restarting pending bulk delete on '%s/%s':%d",
                                 client.getContainerName(), prefix, timestamp));
-                asyncDelete(prefix, timestamp);
+                if (!asyncDelete(prefix, timestamp)) {
+                    deletesToClear.add(prefix);
+                }
+            }
+            if (!deletesToClear.isEmpty()) {
```

Comment:
Scheduled deletes stored in property files would not get removed if the folder was empty.